src/rpc/state_api.rs

```diff
@@ -349,3 +352,28 @@ pub(in crate::rpc) async fn state_fetch_root<DB: Blockstore + Sync + Send + 'sta
 fn lock_pop<T>(mutex: &Mutex<Vec<T>>) -> Option<T> {
     mutex.lock().pop()
 }
+
+/// Get randomness from beacon
+pub(in crate::rpc) async fn state_get_randomness_from_beacon<
+    DB: Blockstore + Send + Sync + 'static,
+>(
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((personalization, rand_epoch, entropy, tsk))): Params<
+        LotusJson<RandomnessParams>,
+    >,
+) -> Result<LotusJson<Vec<u8>>, JsonRpcError> {
+    let state_manager = &data.state_manager;
+    let pts = state_manager.chain_store().load_required_tipset(&tsk)?;
```

Comment:
what is `pts`?

Reply:
`parent_tipset`?

Reply:
yes
https://github.com/filecoin-project/lotus/blob/fbafe1c507f362c82d2cd5540b6c5d580dae8e74/chain/stmgr/stmgr.go#L505C2-L505C56

Reply:
Let's use full names, then. We don't have to copy naming from Lotus, especially confusing abbreviations.

Reply:
corrected